pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/notPepperKit/NotPepperKitCreationService.java

```diff
@@ -0,0 +1,238 @@
+
+        // if the kit type has sub kits > like for testBoston
+//        boolean kitHasSubKits = kitRequestSettings.getHasSubKits() != 0;
+
+        log.info("Setup EasyPost...");
+        EasyPostUtil easyPostUtil = new EasyPostUtil(ddpInstanceDto.getInstanceName());
+
+        if (!checkAddress(juniperKitRequest, kitRequestSettings.getPhone(), easyPostUtil)) {
+            return new KitResponse(ADDRESS_VALIDATION_ERROR, juniperKitRequest.getJuniperKitId());
+        }
+
+        ArrayList<KitRequest> orderKits = new ArrayList<>();
+
+
+        TransactionWrapper.inTransaction(conn -> {
+            createKit(ddpInstance, kitType, juniperKitRequest, kitRequestSettings, easyPostUtil, kitTypeName,  orderKits, conn);
+
+//            only order if external shipper name is set for that kit request, not needed for now
+//            if (StringUtils.isNotBlank(kitRequestSettings.getExternalShipper())) {
+//                return orderExternalKits(kitRequestSettings, orderKits, easyPostUtil, shippingCarrier, conn);
+//            }
+            return null;
+        });
+
+        //send not valid address back to client
+        log.info(juniperKitRequest.getJuniperKitId() + " " + ddpInstance.getName() + " " + kitTypeName + " kit created");
+        return new KitStatus(null, null);
+    }
+
+    /**
+     * checkAddress tries creating an address in EasyPost. If it is successful,
+     * sets the kit's easypostAddressId and returns true, if not returns false
+     * An address is valid only if participant has shortId, first - and lastName, for Juniper shortId is the juniperParticipantId
+     *
+     * @param juniperKitRequest the JuniperKitRequest with address to check
+     */
+
+    public boolean checkAddress(JuniperKitRequest juniperKitRequest, String phone, EasyPostUtil easyPostUtil) {
+        if ((StringUtils.isBlank(juniperKitRequest.getJuniperParticipantID()) && StringUtils.isBlank(juniperKitRequest.getExternalOrderNumber()))
+                || StringUtils.isBlank(juniperKitRequest.getLastName())) {
+            return false;
+        }
+        //let's validate the participant's address
+        String name = "";
+        if (StringUtils.isNotBlank(juniperKitRequest.getFirstName())) {
+            name += juniperKitRequest.getFirstName() + " ";
+        }
+        name += juniperKitRequest.getLastName();
+        if (juniperKitRequest.isSkipAddressValidation()) {
+            try {
+                Address address = easyPostUtil.createBroadAddress(name, juniperKitRequest.getStreet1(), juniperKitRequest.getStreet2(), juniperKitRequest.getCity(),
+                        juniperKitRequest.getPostalCode(), juniperKitRequest.getState(), juniperKitRequest.getCountry(), phone);
+                juniperKitRequest.setEasypostAddressId(address.getId());
+                return true;
+            } catch (EasyPostException e) {
+                log.error("Easypost couldn't create an address for " + juniperKitRequest.getShortId());
```

Comment:
Is it possible this was just a bad request?
If so we should probably not log at the error level.

Reply:
changed to warn